tools/mo/openvino/tools/mo/ops/priorbox_clustered.py

```diff
@@ -72,3 +74,11 @@ def priorbox_clustered_infer(node: Node):
         else:
             res_prod = data_shape[get_height_dim(layout, 4)] * data_shape[get_width_dim(layout, 4)] * num_ratios * 4
             node.out_port(0).data.set_shape([1, 2, res_prod])
+
+    @staticmethod
+    def reverse_infer(node):
+        if node.in_port(0).data.get_shape() is None:
+            node.in_port(0).data.set_shape(undefined_shape_of_rank(1))
+
+        if node.in_port(1).data.get_shape() is None:
+            node.in_port(1).data.set_shape(undefined_shape_of_rank(1))
```

Comment:
you can create helper function for such cases and pass there a list with ranks

Reply:
Done. Added helper function `set_input_shapes` to specify input shapes for image detection ops. 

It works not with just ranks but with partial shapes because for a lot of image detection ops we know not just a rank but a partial shape. An example for `ROIPooling` :
```
set_input_shapes(node, undefined_shape_of_rank(4), shape_array([dynamic_dimension_value, 5]))
```